multiqc/modules/pbmarkdup/pbmarkdup.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from pbmarkdup"""
+
+import logging
+import re
+
+from collections import OrderedDict
+
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """
+    pbmarkdup module class, parses pbmarkdup output.
+    """
+
+    def __init__(self):
+        super(MultiqcModule, self).__init__(
+            name="pbmarkdup",
+            anchor="pbmarkdup",
+            href="https://github.com/PacificBiosciences/pbmarkdup",
+            info=(
+                """
+                 takes one or multiple sequencing chips of an amplified libray
+                 as HiFi reads and marks or removes duplicates.
+                 """
+            ),
+            # Can't find a DOI // doi=
+        )
+
+        self.pbmarkdup = dict()
+
+        for logfile in self.find_log_files("pbmarkdup", filehandles=True):
+            self.pbmarkdup[logfile["s_name"]] = self.parse_logfile(logfile)
+            self.add_data_source(logfile)
+
+        # Filter to strip out ignored sample names
+        self.pbmarkdup = self.ignore_samples(self.pbmarkdup)
+
+        # Raise UserWarning if we did not find any data
+        if not self.pbmarkdup:
+            raise UserWarning
+
+        # Write the parsed data to file
+        self.write_data_file(self.pbmarkdup, "multiqc_pbmarkdup")
+
+        self.pbmarkdup_add_general_stats()
+
+    def parse_logfile(self, logfile):
+        """
+        Parse the standard output from pbmarkdup
+
+        This is currently quite ugly, since pbmarkdup does not have an easily
+        parsable output format. See this github issue:
+        https://github.com/PacificBiosciences/pbbioconda/issues/365
+
+        For now, we just assume that the data fields are always in the same
+        order, without any missing values.
+        """
+
+        file_content = logfile["f"]
+
+        # The number of spaces in the header can vary, based on the length of
+        # the name of the library. We therefore need a regex to match the
+        # header.
+        header_pattern = "LIBRARY *READS *UNIQUE MOLECULES *DUPLICATE READS"
+        pattern = re.compile(header_pattern)
+
+        # The file header
+        file_header = next(file_content).strip()
+
+        # Error if the header doesn't match the expected pattern
+        if not re.match(pattern, file_header):
+            raise RuntimeError(f'Unknown header: "{header}"')
```

Comment:
Please don't raise, just log and skip

```suggestion
            log.error(f"Can't parse file '{logfile["fn"]}', unknown header: '{header}'")
            return False
```

Reply:
Ugh, quotes mess there but you get the idea

Reply:
Added